src/linalg.jl

```diff
@@ -1,19 +1,22 @@
-import Base: +, -, *, /, \
+import Base: +, -, *, /, \, &, |, xor, ~
 
 # TODO: more operators, like AbstractArray
 
 # Unary ops
 @inline -(a::StaticArray) = map(-, a)
+@inline ~(a::StaticArray) = map(~, a)
 
 # Binary ops
 # Between arrays
-@inline +(a::StaticArray, b::StaticArray) = map(+, a, b)
-@inline +(a::AbstractArray, b::StaticArray) = map(+, a, b)
-@inline +(a::StaticArray, b::AbstractArray) = map(+, a, b)
-
-@inline -(a::StaticArray, b::StaticArray) = map(-, a, b)
-@inline -(a::AbstractArray, b::StaticArray) = map(-, a, b)
-@inline -(a::StaticArray, b::AbstractArray) = map(-, a, b)
+let
+    ops = [:+, :-, :&, :|, :xor]
```

Comment:
I think we should probably not do these bitwise ops at all unless they're also part of normal `AbstractArray`.